butane_cli/src/lib.rs

```diff
@@ -92,6 +93,125 @@ pub fn make_migration(base_dir: &Path, name: Option<&String>) -> Result<()> {
     Ok(())
 }
 
+/// Print a description of a column change indented by two spaces.
+pub fn print_column_diff(old: &AColumn, new: &AColumn) -> Result<()> {
+    if old.typeid()? != new.typeid()? {
+        println!("  type: {:?} -> {:?}", old.typeid()?, new.typeid()?);
+    }
+    if old.is_pk() != new.is_pk() {
+        println!("  pk: {} -> {}", old.is_pk(), new.is_pk());
+    }
+    if old.is_auto() != new.is_auto() {
+        println!("  auto: {} -> {}", old.is_auto(), new.is_auto());
+    }
+    if old.nullable() != new.nullable() {
+        println!("  nullable: {} -> {}", old.nullable(), new.nullable());
+    }
+    if old.unique() != new.unique() {
+        println!("  unique: {} -> {}", old.unique(), new.unique());
+    }
+    if old.default() != new.default() {
+        println!("  default: {:?} -> {:?}", old.default(), new.default());
+    }
+    if old.reference() != new.reference() {
+        let old = match old.reference() {
+            Some(ARef::Literal(reference)) => {
+                format!("{}.{}", reference.table_name(), reference.column_name())
+            }
+            None => "None".to_string(),
+            Some(ARef::Deferred(_)) => panic!(),
```

Comment:
Not blocking, but I think I would prefer not panicking here. In practice `Migration::db` resolves all types, and there will probably be other things that break if it doesn't, but the method description/contract doesn't actually promise that

If you keep the panic, I'd prefer adding a panic message indicating why this should be impossible (in this case because `Migration::db` resolves types